Missing equality trait on QImage (and maybe other new types?

This is required because when the type is a `Q_PROPERTY` we generate Rust code that has `==`

### Discussed in https://github.com/KDAB/cxx-qt/discussions/871

<div type='discussions-op-text'>

<sup>Originally posted by **otiv-wannes-vanleemput** February 28, 2024</sup>
I want to visualise a Gstreamer pipeline which runs in a Rust backed. To start off, I would like to update a QImage with the current frame outputted by the pipeline. However, I'm unable to wrap a QImage in a custom rust struct.

This is my cxxqt_object.rs file:

```
#[cxx_qt::bridge]
pub mod qobject {

    unsafe extern "C++" {
        include!("cxx-qt-lib/qimage.h");
        type QImage = cxx_qt_lib::QImage;
    }

    extern "RustQt" {
        #[qobject]
        #[qml_element]
        #[qproperty(QImage, video_frame)]
        type MyObject = super::MyObjectRust;
    }

    impl cxx_qt::Constructor<(QImage,), NewArguments=(QImage,)> for MyObject {}

    unsafe extern "RustQt" {
        #[qinvokable]
        fn video_frame_from_slice(self: Pin<&mut MyObject>, image: &[u8]);
    }
}

use core::pin::Pin;
use cxx_qt_lib::{QImage};
pub struct MyObjectRust {
    video_frame: QImage,
}

impl cxx_qt::Constructor<(QImage,)> for qobject::MyObject {
    type BaseArguments = ();
    type InitializeArguments = ();
    type NewArguments = (QImage,);
    fn route_arguments(args: (QImage, )) -> (Self::NewArguments, Self::BaseArguments, Self::InitializeArguments) {
        (args, (), ())
    }
    fn new((video_frame,): (QImage,)) -> MyObjectRust {
        MyObjectRust {
            video_frame,
        }
    }
}

impl qobject::MyObject {
    pub fn video_frame_from_slice(self: Pin<&mut Self>, image: &[u8]) {
        self.video_frame = QImage::from_data(image, None).unwrap();
    }
}
```

I tried deriving the Default trait, but this resulted in a build error. Thats why I used the constructor.
However, I still get the following build error: 
```
error[E0369]: binary operation `==` cannot be applied to type `cxx_qt_lib::QImage`
 --> src/cxxqt_object.rs:3:1
  |
3 | #[cxx_qt::bridge]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `cxx_qt::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0369`.
```

Should I fork the repository and add the PartialEq trait to QImage, or am I using it incorrectly?</div>

--- a/crates/cxx-qt-lib/src/gui/qfont.rs
+++ b/crates/cxx-qt-lib/src/gui/qfont.rs
@@ -262,6 +262,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qfont_clone"]
         fn construct(font: &QFont) -> QFont;
+
+        #[doc(hidden)]
+        #[rust_name = "qfont_eq"]
+        fn operatorEq(a: &QFont, b: &QFont) -> bool;
     }
 }
 
@@ -290,6 +294,14 @@ impl Clone for QFont {
         ffi::qfont_clone(self)
     }
 }
+
+impl PartialEq for QFont {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qfont_eq(self, other)
+    }
+}
+
+impl Eq for QFont {}
 
 impl QFont {
     /// Returns the bounding rectangle of the current clip if there is a clip;

--- a/crates/cxx-qt-lib/src/gui/qimage.rs
+++ b/crates/cxx-qt-lib/src/gui/qimage.rs
@@ -237,6 +237,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qimage_cache_key"]
         fn qimageCacheKey(image: &QImage) -> i64;
+
+        #[doc(hidden)]
+        #[rust_name = "qimage_eq"]
+        fn operatorEq(a: &QImage, b: &QImage) -> bool;
     }
 }
 
@@ -269,6 +273,14 @@ impl Default for QImage {
         ffi::qimage_init_default()
     }
 }
+
+impl PartialEq for QImage {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qimage_eq(self, other)
+    }
+}
+
+impl Eq for QImage {}
 
 // Safety:
 //

--- a/crates/cxx-qt-lib/src/gui/qpainterpath.rs
+++ b/crates/cxx-qt-lib/src/gui/qpainterpath.rs
@@ -188,6 +188,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qpainterpath_drop"]
         fn drop(painterPath: &mut QPainterPath);
+
+        #[doc(hidden)]
+        #[rust_name = "qpainterpath_eq"]
+        fn operatorEq(a: &QPainterPath, b: &QPainterPath) -> bool;
     }
 }
 
@@ -221,6 +225,14 @@ impl From<&QPointF> for QPainterPath {
         ffi::qpainterpath_from_qpointf(p)
     }
 }
+
+impl PartialEq for QPainterPath {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qpainterpath_eq(self, other)
+    }
+}
+
+impl Eq for QPainterPath {}
 
 // Safety:
 //

--- a/crates/cxx-qt-lib/src/gui/qpen.rs
+++ b/crates/cxx-qt-lib/src/gui/qpen.rs
@@ -113,6 +113,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qpen_clone"]
         fn construct(pen: &QPen) -> QPen;
+
+        #[doc(hidden)]
+        #[rust_name = "qpen_eq"]
+        fn operatorEq(a: &QPen, b: &QPen) -> bool;
     }
 }
 
@@ -142,6 +146,14 @@ impl Clone for QPen {
         ffi::qpen_clone(self)
     }
 }
+
+impl PartialEq for QPen {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qpen_eq(self, other)
+    }
+}
+
+impl Eq for QPen {}
 
 impl From<&ffi::QColor> for QPen {
     fn from(color: &ffi::QColor) -> Self {

--- a/crates/cxx-qt-lib/src/gui/qpolygon.rs
+++ b/crates/cxx-qt-lib/src/gui/qpolygon.rs
@@ -72,6 +72,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qpolygon_clone"]
         fn construct(p: &QPolygon) -> QPolygon;
+
+        #[doc(hidden)]
+        #[rust_name = "qpolygon_eq"]
+        fn operatorEq(a: &QPolygon, b: &QPolygon) -> bool;
     }
 }
 
@@ -106,6 +110,14 @@ impl Clone for QPolygon {
         ffi::qpolygon_clone(self)
     }
 }
+
+impl PartialEq for QPolygon {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qpolygon_eq(self, other)
+    }
+}
+
+impl Eq for QPolygon {}
 
 // Safety:
 //

--- a/crates/cxx-qt-lib/src/gui/qpolygonf.rs
+++ b/crates/cxx-qt-lib/src/gui/qpolygonf.rs
@@ -75,6 +75,10 @@ mod ffi {
         #[doc(hidden)]
         #[rust_name = "qpolygonf_clone"]
         fn construct(p: &QPolygonF) -> QPolygonF;
+
+        #[doc(hidden)]
+        #[rust_name = "qpolygonf_eq"]
+        fn operatorEq(a: &QPolygonF, b: &QPolygonF) -> bool;
     }
 }
 
@@ -109,6 +113,14 @@ impl Clone for QPolygonF {
         ffi::qpolygonf_clone(self)
     }
 }
+
+impl PartialEq for QPolygonF {
+    fn eq(&self, other: &Self) -> bool {
+        ffi::qpolygonf_eq(self, other)
+    }
+}
+
+impl Eq for QPolygonF {}
 
 // Safety:
 //

